tests/AnalyzeProperties.hs

```diff
@@ -0,0 +1,575 @@
+  -- Enforce
+  -- EnforceOne
+  -- KsAuthorized
+  -- NameAuthorized
+  -- Let
+  -- Sequence
+  -- IfThenElse
+genTermSpecific (SizedString _len) = undefined
+  -- Write
+  -- PactVersion
+  -- Format
+  -- FormatTime
+  -- Hash
+  -- Let
+  -- Sequence
+  -- IfThenElse
+
+toPact :: ETerm -> Maybe (Pact.Term Pact.Ref)
+toPact = \case
+  ESimple TDecimal (Inj (DecArithOp op x y)) ->
+    mkApp (arithOpToDef op) [ESimple TDecimal x, ESimple TDecimal y]
+
+  ESimple TDecimal (Inj (DecUnaryArithOp op x)) ->
+    mkApp (unaryArithOpToDef op) [ESimple TDecimal x]
+
+  ESimple TInt (Inj (IntArithOp op x y)) ->
+    mkApp (arithOpToDef op) [ESimple TInt x, ESimple TInt y]
+
+  ESimple TInt (Inj (IntUnaryArithOp op x)) ->
+    mkApp (unaryArithOpToDef op) [ESimple TInt x]
+
+  ESimple TInt (Inj (RoundingLikeOp1 op x)) ->
+    mkApp (roundingLikeOpToDef op) [ESimple TDecimal x]
+
+  ESimple TDecimal (Inj (RoundingLikeOp2 op x y)) ->
+    mkApp (roundingLikeOpToDef op) [ESimple TDecimal x, ESimple TInt y]
+
+  ESimple TDecimal (Inj (DecIntArithOp op x y)) ->
+    mkApp (arithOpToDef op) [ESimple TDecimal x, ESimple TInt y]
+
+  ESimple TDecimal (Inj (IntDecArithOp op x y)) ->
+    mkApp (arithOpToDef op) [ESimple TInt x, ESimple TDecimal y]
+
+  ESimple TInt (Inj (ModOp x y)) ->
+    mkApp modDef [ESimple TInt x, ESimple TInt y]
+
+  ESimple TInt (Inj (StrLength x)) ->
+    mkApp lengthDef [ESimple TStr x]
+
+  ESimple TStr (Inj (StrConcat x y)) ->
+    mkApp addDef [ESimple TStr x, ESimple TStr y]
+
+  ESimple TBool (Inj (IntegerComparison op x y)) ->
+    mkApp (comparisonOpToDef op) [ESimple TInt x, ESimple TInt y]
+
+  ESimple TBool (Inj (DecimalComparison op x y)) ->
+    mkApp (comparisonOpToDef op) [ESimple TDecimal x, ESimple TDecimal y]
+
+  ESimple TBool (Inj (StringComparison op x y)) ->
+    mkApp (comparisonOpToDef op) [ESimple TStr x, ESimple TStr y]
+
+  ESimple TBool (Inj (BoolComparison op x y)) ->
+    mkApp (comparisonOpToDef op) [ESimple TBool x, ESimple TBool y]
+
+  ESimple TBool (Inj (TimeComparison op x y)) ->
+    mkApp (comparisonOpToDef op) [ESimple TTime x, ESimple TTime y]
+
+  ESimple TTime (Inj (IntAddTime x y)) ->
+    mkApp defAddTime [ESimple TTime x, ESimple TInt y]
+
+  ESimple TTime (Inj (DecAddTime x y)) ->
+    mkApp defAddTime [ESimple TTime x, ESimple TDecimal y]
+
+  ESimple TBool (Inj (Logical op args)) ->
+    mkApp (logicalOpToDef op) (ESimple TBool <$> args)
+
+  ESimple TInt     (CoreTerm (Lit x))
+    -> Just $ TLiteral (LInteger x) dummyInfo
+  ESimple TDecimal (CoreTerm (Lit x))
+    -> Just $ TLiteral (LDecimal (unMkDecimal x)) dummyInfo
+  ESimple TStr     (CoreTerm (Lit x))
+    -> Just $ TLiteral (LString (T.pack x)) dummyInfo
+  ESimple TBool    (CoreTerm (Lit x))
+    -> Just $ TLiteral (LBool x) dummyInfo
+  ESimple TTime    (CoreTerm (Lit x))
+    -> Just $ TLiteral (LTime (unMkTime x)) dummyInfo
+
+  ESimple TKeySet  (CoreTerm (Lit _x))
+    -> error "how do we translate keysets?"
```

Comment:
cc @bts

Reply:
Maybe we just generate a different dummy Pact `KeySet` for each separate keyset on the analysis side? Stick a random pubkey in the list, and use the "all" predicate function. When we evaluate the pact vs analysis terms, we randomly pick whether to authorize this keyset. If we do, then on the pact side, supply that dummy pubkey as a valid signatory for the pact execution environment. On the analysis side, just say that keyset is authorized.

Reply:
If we wanted, we could try to do similar things for other aspects of the execution environment. e.g.: does a given key already exist in a db table from a previous tx?